File: stronghold/001_DNA/R/dna.r
#!/usr/bin/env Rscript
# dna.R
# Prints nucleotide counts of a provided DNA sequence through stdin

seq <- readLines(file("stdin"))
nc <- table(strsplit(seq, "")[[1]]) # short for nucleotide counts
writeLines(sprintf("%d %d %d %d", nc[[1]], nc[[2]], nc[[3]], nc[[4]]))

File: stronghold/001_DNA/perl/dna.pl
#!/usr/bin/env perl

sub CountLetters {
  my $seq = $_[0];
  my %counts;

  map {$counts{$_}++} split(//, $_);
  
  return %counts;
}

while (<>) {
  my $seq = $_;
  $seq = uc(chomp($seq));
  my %nuc_counts = CountLetters($seq);

  print(join(" ", (
    $nuc_counts{'A'},
    $nuc_counts{'C'},
    $nuc_counts{'G'},
    $nuc_counts{'T'}
  )), "\n");
}
